src/rulelib.cpp

```diff
@@ -905,7 +858,7 @@ ruleset_print(ruleset_t *rs, rule_t *rules, int detail)
 	int total_support;
 
 	printf("%d rules %d samples\n", rs->n_rules, rs->n_samples);
-	n = (rs->n_samples + BITS_PER_ENTRY - 1) / BITS_PER_ENTRY;
```

Comment:
Is this incorrect?

Reply:
In order to properly print the right number of samples in rule_vector_print, I needed to know the exact amount of samples. Previously, rule_vector_print was only being passed the number of v_entries in non-GMP mode, which would cause it to print 128 samples when there were actually anywhere between 65 and 128. Similarly, if the bitvector had zeroes in the most significant bits, those weren't printed in GMP mode, making the bitvector look shorter than it actually was. So, I had to propagate nsamples through all of the printing functions instead of the number of v_entries.